### DISCUSSION TITLE
Check if there is an _active_pcam at all

### DISCUSSION THREAD

**Discussion by m4rr5** [ramokz/phantom-camera#117] [Ideas]
When you start to setup a scene and add a Camera3D first and then attach a PhantomCameraHost as a child, you end up with a lot of error messages in the log until you add a PhantomCamera3D to your scene too:
res://addons/phantom_camera/scripts/phantom_camera_host/phantom_camera_host.gd:306 - Invalid call. Nonexistent function 'get_global_transform' in base 'Nil'.

I would suggest to add a null check before dereferencing _active_pcam to prevent this (and maybe render a nice warning sign next to the PhantomCameraHost to indicate that you should add at least one PhantomCamera3D to the scene).

Low priority as the fix is to simply add one yourself and the problem disappears. :)

**Comment by ramokz** [comment]
Thanks for spotting this! Should be addressed in the latest hotfix.

Warning sign-wise, the viewfinder (the bottom editor panel called PhantomCamera) does provide a note and a shortcut to add whatever nodes are missing in the scene. But given you raised this, there's an argument to be made that it's not quite enough, so might consider adding something in to improve that piece of UX :)

**Comment by ciso507** [comment]
I think this is similar to what im facing when deleting all phantom cameras and the host, it was breaking my game, i think this fixed it for both pcamera2d and 3d. 👍  And thanks again for this amazing tool. 
func _exit_tree() -> void:
	if is_instance_valid(Properties.pcam_host_owner):

**Reply by ramokz** [reply]
I tried opening a few example scenes and deleting all the `PCams` and the `PCamHost`. Can't seem to trigger any errors or otherwise be prevented from playing a scene — aside from now having a stationary camera.

Are you deleting the `PCams` and `PCamHost` during runtime, or is this all within the editor?

**Reply by ciso507** [reply]
I delete the pcams and the pcamHost with queue_free before queuing free the level and instanciating the new level that has a new set of pcams and pcamHost. But checking if is instance_valid stopped the error i was facing...also i was doing this because my scene was crashing due to godot internal error that was fixed recently about changing scenes. 👍

**Reply by ramokz** [reply]
Sorry for the delayed response!

Just to be sure I'm understanding it correctly.
Is this effectively what you have for the _exit_tree function?

```gdscript
func _exit_tree() -> void:
	if is_instance_valid(Properties.pcam_host_owner):
		Properties.pcam_host_owner.pcam_removed_from_scene(self)

	Properties.pcam_exit_tree(self)
```
